File: 01-Linkedin-Learning/-03-Programming-Foundations-Object-Oriented-design/Course_Code_&_Explanation/5-Class-Diagrams/3-class-diagrams.java

// .           space ship             .
// ___________________________________
// + callSign: String
// - shieldStrength: integer
// ___________________________________
// + fireMissile(): string
// + reduceShield(integer)

public class Spaceship {

  // instance variables
  public string callSign;
  private int shieldStrength;

  //methods
  public string fireMissile() {
    return "pew!";
  }
  public void reduceShield(int amount) {
    shieldStrength -= amount
  }
}

// this is similar to C# 

File: 01-Linkedin-Learning/-03-Programming-Foundations-Object-Oriented-design/Course_Code_&_Explanation/5-Class-Diagrams/3-class-diagrams.py
'''
.           space ship             .
___________________________________
+ callSign: String
- shieldStrength: integer
___________________________________
+ fireMissile(): string
+ reduceShield(integer)
'''

class Spaceship():
  def __init__(self):

    # instance variables 
    self.callSign = ""
    self._shieldStrength = 100 

    # methods 

    def fireMissile(self):
      return "pew!"

    def reduceShield(self, amount):
      self.shieldStrength -= amount


# python doesn't really differentiate between public and private variables. 
# It puts trust in the hands of the programmer not to do anything foolish. 
# So we've added an underscore to the shield strength variable name to indicate that it should be treated as if it were private.